pkg/ext-proc/handlers/server.go

```diff
@@ -62,18 +63,30 @@ func (s *Server) Process(srv extProcPb.ExternalProcessor_ProcessServer) error {
 	// See https://github.com/envoyproxy/envoy/issues/17540.
 	reqCtx := &RequestContext{}
 
+	// Create variable for error handling as each request should only report once for
+	// error metric. This doesn't cover the error "Cannot receive stream request" because
+	// such error might happen even the response is processed.
+	var err error
+	defer func(error) {
+		if reqCtx.ResponseStatusCode != "" {
+			metrics.RecordRequestErrCounter(reqCtx.Model, reqCtx.ResolvedTargetModel, reqCtx.ResponseStatusCode)
+		} else if err != nil {
+			metrics.RecordRequestErrCounter(reqCtx.Model, reqCtx.ResolvedTargetModel, errutil.CanonicalCode(err))
```

Comment:
we will not always have the `Model` or `ResolvedTargetModel` set since some errors may happen before we identify them, correct? is this scenario considered here? are we ok if those two fields are empty?

Reply:
Good point. Having empty fields is fine, however this reminds me that we need to make sure the error code correctly explain why those fields are empty, for example, a `BadRequest` code with empty `Model` makes sense. I reviewed the error codes again and left some comments.

Reply:
Per discussed in other comments, added a new code `BadConfiguration` to cover the cases that model not found.